pkg/controller/appdefinition/pullappimage.go

```diff
@@ -11,9 +12,38 @@ import (
 	"github.com/acorn-io/acorn/pkg/tags"
 	"github.com/acorn-io/baaah/pkg/router"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type imageClient interface {
```

Comment:
This is very much just a stylistic opinion, but consider the readability of this file and what someone looking at it later would want to see first. In other words, what is most important and should be at the top of the file? I would tuck this client at the bottom and keep the actual handler at the top of the file.